src/uu/lsmem/src/lsmem.rs

```diff
@@ -3,15 +3,543 @@
+        if path.is_dir() && filename.starts_with(NAME_MEMORY) {
+            paths.push(path);
+        }
+    }
+    paths
+}
+
+fn is_mergeable(lsmem: &Lsmem, blk: &MemoryBlock) -> bool {
+    if lsmem.nblocks == 0 {
+        return false;
+    }
+
+    let curr_block = &lsmem.blocks[lsmem.nblocks - 1];
+    if lsmem.list_all {
+        return false;
+    }
+    if curr_block.index + curr_block.count != blk.index {
+        return false;
+    }
+    if lsmem.split_by_state && curr_block.state != blk.state {
+        return false;
+    }
+    if lsmem.split_by_removable && curr_block.removable != blk.removable {
+        return false;
+    }
+    if lsmem.split_by_node && lsmem.have_nodes && (curr_block.node != blk.node) {
+        return false;
+    }
+    if lsmem.split_by_zones && lsmem.have_zones {
+        if curr_block.nr_zones != blk.nr_zones {
+            return false;
+        }
+
+        for i in 0..curr_block.nr_zones {
+            if curr_block.zones[i] == ZoneId::ZoneUnknown || curr_block.zones[i] != blk.zones[i] {
+                return false;
+            }
+        }
+    }
+    true
+}
+
+fn memory_block_get_node(path: &PathBuf) -> Result<i32, <i32 as FromStr>::Err> {
+    for entry in fs::read_dir(path).unwrap() {
+        let entry = entry.unwrap();
+        let path = entry.path();
+        let filename = path.to_str().unwrap().split('/').last().unwrap();
+        if path.is_dir() && filename.starts_with("node") {
+            return filename["node".len()..].parse();
+        }
+    }
+    Ok(-1)
+}
+
+fn memory_block_read_attrs(lsmem: &Lsmem, path: &PathBuf) -> MemoryBlock {
+    let mut blk = MemoryBlock::new();
+    blk.count = 1;
+    blk.state = MemoryState::Unknown;
+    let filename = path.to_str().unwrap().split('/').last().unwrap();
+    blk.index = filename[NAME_MEMORY.len()..].parse().unwrap();
+
+    let mut removable_path = path.clone();
+    removable_path.push(PATH_SUB_REMOVABLE);
+    blk.removable = read_file_content::<i32>(&removable_path).is_ok();
+
+    let mut state_path = path.clone();
+    state_path.push(PATH_SUB_STATE);
+    if let Ok(state_raw) = read_file_content::<String>(&state_path) {
+        blk.state = MemoryState::from_str(&state_raw).unwrap();
+    }
+
+    if lsmem.have_nodes {
+        blk.node = memory_block_get_node(path).unwrap();
+    }
+
+    blk.nr_zones = 0;
+    if lsmem.have_zones {
+        if let Ok(raw_content) = read_file_content::<String>(Path::new(PATH_VALID_ZONES)) {
+            let zone_toks = raw_content.split(' ').collect::<Vec<&str>>();
+            for (i, zone_tok) in zone_toks
+                .iter()
+                .enumerate()
+                .take(std::cmp::min(zone_toks.len(), ZoneId::MaxNrZones as usize))
+            {
+                blk.zones[i] = serde_json::from_str(zone_tok).unwrap();
+                blk.nr_zones += 1;
+            }
+        }
+    }
+    blk
+}
+
+fn set_summary(lsmem: &Lsmem) -> Vec<TableRow> {
```

Comment:
This function name is a bit misleading as the function is unrelated to `print_summary()`. Instead, the function is related to `print_table` and creates a table. So I suggest to change the return type to `Table` and name the function something like `create_table`.

Reply:
fixed